packages/yew-macro/src/html_tree/lint/mod.rs

```diff
@@ -0,0 +1,94 @@
+//! Lints to catch possible misuse of the `html!` macro use. At the moment these are mostly focused
+//! on accessibility.
+
+use syn::spanned::Spanned;
+
+use crate::props::{ElementProps, Prop};
+
+use super::{html_element::HtmlElement, HtmlTree};
+
+pub trait Lint {
+    fn lint(element: &HtmlElement);
+}
+
+pub fn lint_all(tree: &HtmlTree) {
+    lint::<AHrefLint>(tree);
+    lint::<ImgAltLint>(tree);
+}
+
+pub fn lint<L>(tree: &HtmlTree)
+where
+    L: Lint,
+{
+    match tree {
+        HtmlTree::List(list) => {
+            for child in &list.children.0 {
+                lint::<L>(&child)
+            }
+        }
+        HtmlTree::Element(el) => L::lint(&el),
+        _ => {}
+    }
+}
+
+fn get_attribute<'a>(props: &'a ElementProps, name: &str) -> Option<&'a Prop> {
+    props
+        .attributes
+        .iter()
+        .find(|item| item.label.to_ascii_lowercase_string() == name)
+}
+
+pub struct AHrefLint;
+
+impl Lint for AHrefLint {
+    fn lint(element: &HtmlElement) {
+        let tag_name = &element.name;
+        if let super::html_element::TagName::Lit(ref lit) = element.name {
+            if lit.to_ascii_lowercase_string() != "a" {
+                return;
+            };
+            if let Some(prop) = get_attribute(&element.props, "href") {
+                if let syn::Expr::Lit(lit) = &prop.value {
+                    if let syn::Lit::Str(str) = &lit.lit {
+                        if str.value() == "#" {
+                            emit_warning!(
+                                quote::quote! {#lit}.span(),
```

Comment:
Isn't this just:
```suggestion
                                lit.span(),
```
?

Reply:
Yes :D